messaging/src/main/scala/uk/ac/wellcome/messaging/sqsworker/alpakka/AlpakkaSQSWorker.scala

```diff
@@ -12,24 +12,31 @@ import uk.ac.wellcome.messaging.worker._
 import uk.ac.wellcome.messaging.worker.models._
 import uk.ac.wellcome.messaging.worker.steps.MonitoringProcessor
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
-class AlpakkaSQSWorker[Work, MonitoringContext, Summary](
-  config: AlpakkaSQSWorkerConfig
+class AlpakkaSQSWorker[Work,
+                       InfraServiceMonitoringContext,
```

Comment:
What are the multiple contexts for?

Reply:
It's because what we want to pass around within each service and what we want to pass around between services is different. For example for Opentracing, what we want to pass around within a service is a `Span`which is a mutable thing that offers various operations like tagging errors etc, and most importantly it allows to call `finish` to mark a span as complete.
What we want to pass around between services is a serialization of. a `SpanContext`, probably in teh form of a `Map[String, String]`. A `SpanContext` identifies a `Span`  and for example, allows to create another `Span` as a child of the one identified by `SpanContext`. But there is no way of retrieving the `Span` out of a `SpanContext`.
So within a service we want a `Span` cause that allows us to mutate it and to `finish` it. The `SpanContext` is the serializable thing that we are going to pass around in SNS messages